fractal-puppy/Engine
# Move UI interaction to KeyUP

- Moved UI interaction logic from isPressed (key down) to key up. 
- Corrected some hover checks: pressing mouse 1 checked if ui flag for mouse 3 in some cases 
  - *Unimportant at the moment since all placed ui elements raise hover flags for both mouse 1 and mouse 3 (ui elements have the option to set a "hovering ui element" flad for mouse 1 and mouse 3 independently. By default, ui elements report the flag for both and all placed elements I have seen use the default behaviour and, thus, both flags are the same)*

---

- The player's character should not move when interacting with a ui element
- Bug was specially reported when inteacting with the skill assigment popup

**Test**
- Open the skills menu and unlock some skills
- Close the skills menu
- Open the skill assignment popup (right-click on one of the ability slots
- Assign a skill (Player should not move)

- Additionally: Test opening and closing some other menus and popups and check that the player doesn't move
